databuilder/extractor/snowflake_metadata_extractor.py

```diff
@@ -0,0 +1,148 @@
+import logging
+from collections import namedtuple
+
+from pyhocon import ConfigFactory, ConfigTree  # noqa: F401
+from typing import Iterator, Union, Dict, Any  # noqa: F401
+from unidecode import unidecode
+
+from databuilder import Scoped
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.sql_alchemy_extractor import SQLAlchemyExtractor
+from databuilder.models.table_metadata import TableMetadata, ColumnMetadata
+from itertools import groupby
+
+
+TableKey = namedtuple('TableKey', ['schema_name', 'table_name'])
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SnowflakeMetadataExtractor(Extractor):
+    """
+    Extracts Snowflake table and column metadata from underlying meta store database using SQLAlchemyExtractor.
+    Requirements:
+        snowflake-connector-python
+        snowflake-sqlalchemy
+    """
+    # SELECT statement from snowflake information_schema to extract table and column metadata
+    SQL_STATEMENT = """
+    SELECT
+        lower(c.column_name) AS col_name,
+        c.comment AS col_description,
+        lower(c.data_type) AS col_type,
+        lower(c.ordinal_position) AS col_sort_order,
+        lower(c.table_catalog) AS database,
+        lower({cluster_source}) AS cluster,
+        lower(c.table_schema) AS schema_name,
+        lower(c.table_name) AS name,
+        t.comment AS description,
+        decode(lower(t.table_type), 'view', 'true', 'false') AS is_view
+    FROM
+        {database}.INFORMATION_SCHEMA.COLUMNS AS c
+    LEFT JOIN
+        {database}.INFORMATION_SCHEMA.TABLES t
+            ON c.TABLE_NAME = t.TABLE_NAME
+            AND c.TABLE_SCHEMA = t.TABLE_SCHEMA
+    {where_clause_suffix};
+    """
+
+    # CONFIG KEYS
+    WHERE_CLAUSE_SUFFIX_KEY = 'where_clause_suffix'
+    CLUSTER_KEY = 'cluster_key'
+    USE_CATALOG_AS_CLUSTER_NAME = 'use_catalog_as_cluster_name'
+    DATABASE_KEY = 'database_key'
+
+    # Default values
+    DEFAULT_CLUSTER_NAME = 'master'
+
+    DEFAULT_CONFIG = ConfigFactory.from_dict(
+        {WHERE_CLAUSE_SUFFIX_KEY: ' ', CLUSTER_KEY: DEFAULT_CLUSTER_NAME, USE_CATALOG_AS_CLUSTER_NAME: True}
+    )
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+        conf = conf.with_fallback(SnowflakeMetadataExtractor.DEFAULT_CONFIG)
+        self._cluster = '{}'.format(conf.get_string(SnowflakeMetadataExtractor.CLUSTER_KEY))
+
+        if conf.get_bool(SnowflakeMetadataExtractor.USE_CATALOG_AS_CLUSTER_NAME):
```

Comment:
May be we can have a default value on this config?

Reply:
ok

Reply:
@jinhyukchang, but there is default value, if you mean default value for param `USE_CATALOG_AS_CLUSTER_NAME ` or did you mean something else?

Reply:
> @jinhyukchang, but there is default value, if you mean default value for param `USE_CATALOG_AS_CLUSTER_NAME ` in `DEFAULT_CONFIG` dict or did you mean something else?

NVM. Sorry, seems I missed it.